docs/src/dev-guide/design-kv-ir-streams/background.md

```
@@ -0,0 +1,318 @@
+with the goal of each encoding method being to deduplicate any repetitive information (e.g.,
+deduplicating repeated `VarString`s with a dictionary) and then represent the value with a 64-bit
+integer. For each encoded value, clp-s stores it in the value's corresponding ERT column. ERTs are
+efficient to search since all the values are integers; and by grouping events with the same schema
+into an ERT, clp-s essentially deduplicates the event's schema.
+
+clp-s encodes each leaf-node value type as follows:
+
+* `Integer`s are encoded natively.
+* `Float`s are encoded using the IEEE-754 double-precision format.
+* `Boolean`s are encoded as an integer.
+* `VarString`s are encoded using dictionary-encoding---i.e., a dictionary that maps string values to
```

markdownlint-cli2 (0.17.2)

Line length
Expected: 80; Actual: 100

(MD013, line-length)